lib/manager/gomod/artifacts.ts

```diff
@@ -68,6 +69,21 @@ function getUpdateImportPathCmds(
   return updateImportCommands;
 }
 
+function useModcacherw(goVersion: string): boolean {
+  if (!is.string(goVersion)) {
+    return true;
+  }
+
+  const [majorPart, minorPart] = goVersion.replace(/^[^\d]*/g, '').split('.');
```

Comment:
`/^[^\d]*/g` won't this remove all dots? Isn't this simpler?
```suggestion
  const [,majorPart, minorPart] = goVersion.match(/(\d+)\.(\d+)/) ?? [];
```